imageproxy.go

```diff
@@ -162,7 +168,15 @@ func (p *Proxy) serveImage(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	copyHeader(w.Header(), resp.Header, "Content-Length", "Content-Type")
+	if contentType := p.allowedContentType(resp.Header.Get("Content-Type")); contentType != "" {
+		w.Header().Set("Content-Type", contentType)
+		w.Header().Set("X-Content-Type-Options", "nosniff")
```

Comment:
Doesn't https://github.com/whatwg/fetch/issues/395 suggest that you shouldn't use nosniff with images?

And can this content type check simply be moved into the existing allowed() method?

Reply:
My current understanding is that it's fine (and more secure) as long you also specify the correct Content-Type. I should research and test that a bit more though.

I think the reason it's not in the existing `allowed` method is because it has to happen _after_ we fetch the remote image.